usvg/src/image.rs

```diff
@@ -111,34 +130,45 @@ pub(crate) fn convert(
     Some(())
 }
 
-pub(crate) fn get_href_data(
-    element_id: &str,
-    href: &str,
-    opt: &OptionsRef,
-) -> Option<ImageKind> {
+pub(crate) fn get_href_data(href: &str, opt: &OptionsRef) -> Option<ImageKind> {
     if let Ok(url) = data_url::DataUrl::process(href) {
         let (data, _) = url.decode_to_vec().ok()?;
-        match (url.mime_type().type_.as_str(), url.mime_type().subtype.as_str()) {
-            ("image", "jpg") | ("image", "jpeg") => Some(ImageKind::JPEG(data)),
-            ("image", "png") => Some(ImageKind::PNG(data)),
-            ("image", "svg+xml") => load_sub_svg(&data, opt),
-            ("text", "plain") => {
-                match get_image_data_format(&data) {
-                    Some(ImageFormat::JPEG) => {
-                        Some(ImageKind::JPEG(data))
-                    }
-                    Some(ImageFormat::PNG) => {
-                        Some(ImageKind::PNG(data))
-                    }
-                    _ => {
-                        load_sub_svg(&data, opt)
-                    }
-                }
-            }
-            _ => None,
-        }
+
+        let mime = format!(
+            "{}/{}",
+            url.mime_type().type_.as_str(),
+            url.mime_type().subtype.as_str()
+        );
+
+        (opt.image_href_resolver.resolve_data)(&mime, Arc::new(data), opt)
     } else {
-        let path = opt.get_abs_path(std::path::Path::new(href));
+        (opt.image_href_resolver.resolve_string)(href, opt)
+    }
+}
+
+/// Create DataUrl resolver function that handles standard mime types for JPEG, PNG and SVG.
+#[allow(clippy::type_complexity)]
+pub fn create_default_data_resolver<'a>() -> Box<dyn Fn(&str, Arc<Vec<u8>>, &OptionsRef) -> Option<ImageKind> + 'a> {
```

Comment:
I would prefer those methods to be part of `ImageHrefResolver` as static one. So we would not clatter the global namespace.

And I would rename them to something like `ImageHrefResolver::default_data_resolver`. I don't think we need create/new prefix here.